java-spiffe-core/src/main/java/io/spiffe/svid/jwtsvid/JwtSvid.java

```diff
@@ -0,0 +1,254 @@
+            val signedJwt = SignedJWT.parse(token);
+            val claimsSet = signedJwt.getJWTClaimsSet();
+
+            List<String> claimAudience = claimsSet.getAudience();
+            validateAudience(claimAudience, audience);
+
+            val expirationTime = claimsSet.getExpirationTime();
+            validateExpiration(expirationTime);
+
+            val spiffeId = getSpiffeId(claimsSet);
+            val jwtBundle = jwtBundleSource.getBundleForTrustDomain(spiffeId.getTrustDomain());
+
+            val keyId = getKeyId(signedJwt.getHeader());
+            val jwtAuthority = jwtBundle.findJwtAuthority(keyId);
+
+            val algorithm = signedJwt.getHeader().getAlgorithm().getName();
+            verifySignature(signedJwt, jwtAuthority, algorithm, keyId);
+
+            return new JwtSvid(spiffeId, claimAudience, expirationTime, claimsSet.getClaims(), token);
+        } catch (ParseException e) {
+            throw new IllegalArgumentException("Unable to parse JWT token", e);
+        }
+    }
+
+    /**
+     * Parses and validates a JWT-SVID token and returns an instance of a {@link JwtSvid}.
+     * <p>
+     * The JWT-SVID signature is not verified.
+     *
+     * @param token    a token as a string that is parsed and validated
+     * @param audience audience as a list of strings used to validate the 'aud' claim
+     * @return an instance of a {@link JwtSvid} with a SPIFFE ID parsed from the 'sub', audience from 'aud', and expiry
+     * from 'exp' claim.
+     * @throws JwtSvidException when the token expired or the expiration claim is missing, or when
+     *                                           the 'aud' has an audience that is not in the audience provided as parameter
+     * @throws IllegalArgumentException          when the token cannot be parsed
+     */
+    public static JwtSvid parseInsecure(@NonNull final String token, @NonNull final List<String> audience) throws JwtSvidException {
+        if (StringUtils.isBlank(token)) {
+            throw new IllegalArgumentException("Token cannot be blank");
+        }
+
+        try {
+            val signedJwt = SignedJWT.parse(token);
+            val claimsSet = signedJwt.getJWTClaimsSet();
+
+            List<String> claimAudience = claimsSet.getAudience();
+            validateAudience(claimAudience, audience);
+
+            val expirationTime = claimsSet.getExpirationTime();
+            validateExpiration(expirationTime);
+
+            val spiffeId = getSpiffeId(claimsSet);
+
+            return new JwtSvid(spiffeId, claimAudience, expirationTime, claimsSet.getClaims(), token);
+        } catch (ParseException e) {
+            throw new IllegalArgumentException("Unable to parse JWT token", e);
+        }
+    }
+
+    /**
+     * Returns the JWT-SVID marshaled to a string. The returned value is the same token value originally passed
+     * to the parseAndValidate method.
+     *
+     * @return the token as String
+     */
+    public String marshall() {
```

Comment:
Not sure if `marshall` is a correct spelling, I'm more accustomed to seeing it spelled `marshal`

Reply:
yes, it's a typo